File: src/features/typing/data/go/daily_temperatures.go
package main

func dailyTemperatures(temperatures []int) []int {
	result := make([]int, len(temperatures))
	stack := []int{}
	for i, t := range temperatures {
		for len(stack) > 0 && temperatures[stack[len(stack)-1]] < t {
			idx := stack[len(stack)-1]
			stack = stack[:len(stack)-1]
			result[idx] = i - idx
		}
		stack = append(stack, i)
	}
	return result
}

File: src/features/typing/data/dart/daily_temperatures.dart
class Solution {
  List<int> dailyTemperatures(List<int> temperatures) {
    List<int> result = List.filled(temperatures.length, 0);
    List<int> stack = [];
    for (var i = 0; i < temperatures.length; i++) {
      int t = temperatures[i];
      while (stack.isNotEmpty && temperatures[stack.last] < t) {
        int idx = stack.removeLast();
        result[idx] = i - idx;
      }
      stack.add(i);
    }
    return result;
  }
}
